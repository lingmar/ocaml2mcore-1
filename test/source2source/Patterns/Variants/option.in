type 'a option =
  | None
  | Some of 'a

let get_or d = function
  | None -> d
  | Some v -> v

let _ =
  let n = None in
  let s = Some 1 in
  print_endline (string_of_int (get_or 42 s));
  print_endline (string_of_int (get_or 42 n));
  print_endline (Char.escaped (get_or 'a' (Some 'B')))
