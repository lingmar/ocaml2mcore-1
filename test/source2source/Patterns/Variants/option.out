1
42
B
